traffic_portal/test/integration/specs/Profiles.spec.ts

```diff
@@ -33,47 +33,57 @@ describe('Setup API for Profiles', () => {
     it('Setup', async () => {
         await api.UseAPI(profiles.setup);
     });
-});
-profiles.tests.forEach(async profilesData => {
-    profilesData.logins.forEach(login => {
-        describe(`Traffic Portal - Profiles - ${login.description}`, () => {
-            it('can login', async () => {
-                browser.get(browser.params.baseUrl);
-                await loginPage.Login(login);
-                expect(await loginPage.CheckUserName(login)).toBeTruthy();
-            });
-            it('can open profiles page', async () => {
-                await profilesPage.OpenConfigureMenu();
-                await profilesPage.OpenProfilesPage();
-            });
-            profilesData.add.forEach(add => {
-                it(add.description, async () => {
-                    expect(await profilesPage.CreateProfile(add)).toBeTruthy();
+    profiles.tests.forEach(async profilesData => {
+        profilesData.logins.forEach(login => {
```

Comment:
The restructuring here confuses me - this used to be a separate `describe` for each login for each profile, but now it's all under the setup `describe`? It looks that way, but then the `describe`'s description isn't accurate, it's not just setting up it's actually doing all the Profile testing. That's not a terrible way to do it, but given that these specs are asynchronous, there's no guarantee that any one of them will run before any other (and in general you shouldn't assume that to allow for greater degrees of parallelism); that is to say, you don't know that the setup is run to completion before the other tests run. So it should probably go in a `beforeAll` - I actually think it ought to have been done that way in the first place, but the subsequent suite method was working so that change was unnecessary until the `it`s were sharing a `describe`.

By the same reasoning, the teardown logic in the "cleanup" spec should also probably be in an `afterAll`.

Further, we're re-adding `afterEach` `describe`, but it can actually just be added once in the outer `describe` since all spec blocks in a nested suite are also in the outer suite, and so are affected by `beforeEach` and `afterEach` in that outer block.

Reply:
I will fix the structure back to 3 describes. The **beforeAll** is a good idea, I will add that in.  I have to test this out for the **afterall** to answer you because I read somewhere that you can only have one **afterall** and I already use that for the log out function (log out 3 times for each users). @ocket8888

Reply:
You can have one `afterAll` per `describe`, I think - but also you can just do multiple things in one `afterAll` if that's what you're talking about.